### PR TITLE
Cleanup float

### PR DESCRIPTION
**Pull request by wutschel**
Major rework to use correct types instead of float.

- Covers `CGRectMake`, `CGPointMake`, `CGSizeMake`, `UIEdgeInstetsMake`, `minimumScaleFactor`, `cornerRadius`, `borderWidth`, `shadowRadius`, `alpha` and related variables and defines
- Use `NSTimeInterval` type variables and do not combine with `float` or `f` suffix
- Remove unused variable `storeBrightness`

### CODE REVIEW

**Comment by wutschel** [comment]
@kambala-decapitator, thanks for merging the other PRs. Now I could prepare the `float`-rework. There are now only very few `float` or `f` suffixes left. The majority is related to `UIColor` definitions, which I want to resolve in a dedicated PR which will then also implement Dark Mode.

File: XBMC Remote/DetailViewController.m [hunk 1]
```diff
@@ -5326,10 +5326,10 @@ -(void)setIpadInterface:(NSDictionary *)itemSizes{
     albumFontSize = 18;
     trackCountFontSize = 13;
     labelPadding = 8;
-    cellGridWidth =117.0f;
-    cellGridHeight =  168.0f;
-    fullscreenCellGridWidth = 164.0f;
-    fullscreenCellGridHeight = 246.0f;
+    cellGridWidth =117;
+    cellGridHeight =  168;
```
Comment (kambala-decapitator) [inline XBMC Remote/DetailViewController.m:5330]: please fix whitespace as well

File: XBMC Remote/HostViewController.m [hunk 2]
```diff
@@ -46,7 +46,7 @@ - (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
     self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
     return self;
 }
-- (void)AnimLabel:(UIView *)Lab AnimDuration:(float)seconds Alpha:(float)alphavalue XPos:(int)X{
+- (void)AnimLabel:(UIView *)Lab AnimDuration:(NSTimeInterval)seconds Alpha:(CGFloat)alphavalue XPos:(int)X{
```
Comment (kambala-decapitator) [inline XBMC Remote/HostViewController.m:49]: these animation methods look quite similar to those above, probably should be extracted to UIView category or some utility function

File: XBMC Remote/KenBurns/JBKenBurnsView.m [hunk 3]
```diff
@@ -159,17 +159,17 @@ - (void) _animate:(NSNumber*)num{
...
+    CGFloat widthDiff     = -1;
+    CGFloat heightDiff    = -1;
+    CGFloat originX       = -1;
+    CGFloat originY       = -1;
+    CGFloat zoomInX       = -1;
+    CGFloat zoomInY       = -1;
+    CGFloat moveX         = -1;
+    CGFloat moveY         = -1;
+    CGFloat frameWidth    = isLandscape? self.frame.size.width : self.frame.size.height;
+    CGFloat frameHeight   = isLandscape? self.frame.size.height : self.frame.size.width;
```
Comment (kambala-decapitator) [inline XBMC Remote/KenBurns/JBKenBurnsView.m:172]: please add leading whitespace to `?` as well

File: XBMC Remote/KenBurns/JBKenBurnsView.m [hunk 4]
```diff
@@ -230,15 +230,15 @@ - (void) _animate:(NSNumber*)num{
...
     imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, optimusWidth, optimusHeight)];
     
     // Calcule the maximum move allowed.
-    float maxMoveX = optimusWidth - frameWidth;
-    float maxMoveY = optimusHeight - frameHeight;
+    CGFloat maxMoveX = optimusWidth - frameWidth;
+    CGFloat maxMoveY = optimusHeight - frameHeight;
     
-    float rotation = (arc4random() % 9) / 100;
+    CGFloat rotation = (arc4random() % 9) / 100;
```
Comment (kambala-decapitator) [inline XBMC Remote/KenBurns/JBKenBurnsView.m:241]: FYI `arc4random_uniform()` is preferred over `arc4random() % X`

File: XBMC Remote/NowPlaying.m [hunk 5]
```diff
@@ -270,7 +270,7 @@ -(void)fadeView:(UIView *)view hidden:(BOOL)value{
     view.hidden=value;
 }
 
-- (void)AnimTable:(UITableView *)tV AnimDuration:(float)seconds Alpha:(float)alphavalue XPos:(int)X{
+- (void)AnimTable:(UITableView *)tV AnimDuration:(NSTimeInterval)seconds Alpha:(CGFloat)alphavalue XPos:(int)X{
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:273]: familiar animation again :)

File: XBMC Remote/PosterCell.m [hunk 6]
```diff
@@ -19,8 +19,8 @@ @implementation PosterCell
 - (id)initWithFrame:(CGRect)frame{
     self = [super initWithFrame:frame];
     if (self) {
-        float labelHeight = (int)(frame.size.height * 0.19f);
-        float borderWidth = [self halfSizeIfRetina:1.0f];
+        CGFloat labelHeight = (CGFloat)(frame.size.height * 0.19);
```
Comment (kambala-decapitator) [inline XBMC Remote/PosterCell.m:22]: 1. you've removed rounding down
2. actually I suggest to use rounding up instead

File: XBMC Remote/RightMenuViewController.m [hunk 7]
```diff
@@ -93,8 +93,8 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
         UIView *backView = [[UIView alloc] initWithFrame:cell.frame];
         [backView setBackgroundColor:[UIColor colorWithRed:.086 green:.086 blue:.086 alpha:1]];
         cell.selectedBackgroundView = backView;
-        UIImageView *xbmc_logo = [[UIImageView alloc] initWithFrame:CGRectMake(self.view.bounds.size.width - 195.0f, (int)((44/2) - (36/2)) - 2, 145, 36)];
-        xbmc_logo. alpha = .25f;
+        UIImageView *xbmc_logo = [[UIImageView alloc] initWithFrame:CGRectMake(self.view.bounds.size.width - 195, (int)((44/2) - (36/2)) - 2, 145, 36)];
```
Comment (kambala-decapitator) [inline XBMC Remote/RightMenuViewController.m:96]: I'm seeing this magical computation code for the second time ;)

File: XBMC Remote/SDWebImage/SDWebImageDownloaderOperation.m [hunk 8]
```diff
@@ -230,7 +230,7 @@ - (void)connection:(NSURLConnection *)connection didReceiveData:(NSData *)data
                     CGColorSpaceRelease(colorSpace);
                     if (bmContext)
                     {
-                        CGContextDrawImage(bmContext, (CGRect){.origin.x = 0.0f, .origin.y = 0.0f, .size.width = self->width, .size.height = partialHeight}, partialImageRef);
+                        CGContextDrawImage(bmContext, (CGRect){.origin.x = 0, .origin.y = 0, .size.width = self->width, .size.height = partialHeight}, partialImageRef);
```
Comment (kambala-decapitator) [inline XBMC Remote/SDWebImage/SDWebImageDownloaderOperation.m:233]: please replace with `CGRectMake`

File: XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m [hunk 9]
```diff
@@ -550,7 +550,7 @@ - (void)drawRect:(CGRect)rect {
 	
 	// Gradient Declaration
 	CGColorSpaceRef colorSpace = CGColorSpaceCreateDeviceRGB();
-	CGFloat alphaGradientLocations[] = {0, 0.8f};
+	CGFloat alphaGradientLocations[] = {0, 0.8};
     
 	CGGradientRef alphaGradient = nil;
     if([[[UIDevice currentDevice] systemVersion]floatValue] >= 5){
```
Comment (kambala-decapitator) [inline XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m:556]: we need to get rid of such useless things

File: XBMC Remote/Utilities.m [hunk 10]
```diff
@@ -107,8 +107,8 @@ - (UIImage*)colorizeImage:(UIImage *)image withColor:(UIColor*)color{
     UIGraphicsBeginImageContextWithOptions(image.size, YES, [[UIScreen mainScreen] scale]);
     
     CGRect contextRect;
-    contextRect.origin.x = 0.0f;
-    contextRect.origin.y = 0.0f;
+    contextRect.origin.x = 0;
+    contextRect.origin.y = 0;
     contextRect.size = [image size];
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:112]: this is a good candidate for using `struct` initialiser :)

File: XBMC Remote/iPad/MenuViewController.m [hunk 11]
```diff
@@ -242,7 +242,7 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
             int cellHeight = PAD_MENU_INFO_HEIGHT;
             int cellHeightPad = cellHeight - 4;
             UIImageView *xbmc_logo = [[UIImageView alloc] initWithFrame:CGRectMake(232, (int)((cellHeight/2) - (cellHeightPad/2)) - 1, 73, cellHeightPad)];
-            xbmc_logo. alpha = .25f;
+            xbmc_logo. alpha = 0.25;
```
Comment (kambala-decapitator) [inline XBMC Remote/iPad/MenuViewController.m:245]: please fix whitespace as well

File: XBMC Remote/iPad/StackScrollViewController.m [hunk 12]
```diff
@@ -131,9 +131,9 @@ -(void)handleStackScrollFullScreenEnabled:(NSNotification *)sender{
     }
     BOOL hideToolbar = [[[sender userInfo] valueForKey:@"hideToolbar"] boolValue];
     BOOL clipsToBounds = [[[sender userInfo] valueForKey:@"clipsToBounds"] boolValue];
-    float duration = [[[sender userInfo] valueForKey:@"duration"] floatValue];
+    NSTimeInterval duration = [[[sender userInfo] valueForKey:@"duration"] floatValue];
```
Comment (kambala-decapitator) [inline XBMC Remote/iPad/StackScrollViewController.m:134]: 1. use `doubleValue`
2. FYI generally `-valueForKey:` shouldn't be used to get dictionary value

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
damn I made it

**Comment by wutschel** [comment]
> damn

:) Very much appreciated. Will walk through your comments now.

File: XBMC Remote/DetailViewController.m [hunk 13]
```diff
@@ -2410,12 +2410,12 @@ - (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger
                                   andResize:CGSizeMake(albumThumbHeight, albumThumbHeight)
                                   completed:^(UIImage *image, NSError *error, SDImageCacheType cacheType) {
                                       BOOL isRetina = ([[UIScreen mainScreen] respondsToSelector:@selector(scale)] && [[UIScreen mainScreen] scale] >= 2);
-                                      float thumbBorder = isRetina ? 1.0f/[[UIScreen mainScreen] scale] : 1.0f;
+                                      CGFloat thumbBorder = isRetina ? 1.0/[[UIScreen mainScreen] scale] : 1.0;
```
[the remark this answers is not in the document]
Comment (wutschel) [reply XBMC Remote/DetailViewController.m:2413]: Ok, what about `isRetinaHD`?

File: XBMC Remote/iPad/StackScrollViewController.m (hunk 12 above)
Comment (wutschel) [reply XBMC Remote/iPad/StackScrollViewController.m:134]: Question: There are still several cases where `floatValue` is used. I understand the variable type is relevant for the output (= the variable type the value assigned to). Is this understanding correct?

File: XBMC Remote/DetailViewController.m [hunk 14]
```diff
@@ -1243,17 +1243,17 @@ -(CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollectionV
...
         }
         else {
-            [flowLayout setMinimumLineSpacing:4.0f];
+            [flowLayout setMinimumLineSpacing:4];
         }
         [flowLayout setMinimumInteritemSpacing:cellMinimumLineSpacing];
     }
     else{
-        if (!cellMinimumLineSpacing) cellMinimumLineSpacing = 0.0f;
+        if (!cellMinimumLineSpacing) cellMinimumLineSpacing = 0;
```
[the remark this answers is not in the document]
Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: Done with d4d0dd2.

File: XBMC Remote/DetailViewController.m [hunk 15]
```diff
@@ -2410,12 +2410,12 @@ - (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger
...
                                   completed:^(UIImage *image, NSError *error, SDImageCacheType cacheType) {
                                       BOOL isRetina = ([[UIScreen mainScreen] respondsToSelector:@selector(scale)] && [[UIScreen mainScreen] scale] >= 2);
-                                      float thumbBorder = isRetina ? 1.0f/[[UIScreen mainScreen] scale] : 1.0f;
+                                      CGFloat thumbBorder = isRetina ? 1.0/[[UIScreen mainScreen] scale] : 1.0;
                                       [thumbImageContainer setBackgroundColor:[UIColor clearColor]];
                                       thumbImageContainer.layer.shadowColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:1].CGColor;
                                       thumbImageContainer.layer.shadowOpacity = 1.0f;
-                                      thumbImageContainer.layer.shadowOffset = CGSizeMake(0.0f, 0.0f);
-                                      thumbImageContainer.layer.shadowRadius = 2.0f;
+                                      thumbImageContainer.layer.shadowOffset = CGSizeMake(0, 0);
```
[the remark this answers is not in the document]
Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: Done with 1e3e046.

File: XBMC Remote/DetailViewController.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: Done with 1d8f4bc.

File: XBMC Remote/iPad/MenuViewController.m (hunk 11 above)
Comment (wutschel) [reply XBMC Remote/iPad/MenuViewController.m]: Done with 1d8f4bc.

File: XBMC Remote/Utilities.m (hunk 10 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m]: Done with abb8cee.

File: XBMC Remote/SDWebImage/SDWebImageDownloaderOperation.m (hunk 8 above)
Comment (wutschel) [reply XBMC Remote/SDWebImage/SDWebImageDownloaderOperation.m]: Done with dd2cc9b.

File: XBMC Remote/KenBurns/JBKenBurnsView.m (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/KenBurns/JBKenBurnsView.m]: Done with 0555baf.

File: XBMC Remote/PosterCell.m (hunk 6 above)
Comment (wutschel) [reply XBMC Remote/PosterCell.m]: Done with b794d5b.

File: XBMC Remote/RightMenuViewController.m (hunk 7 above)
Comment (wutschel) [reply XBMC Remote/RightMenuViewController.m:96]: Oh, I see it even three times, with a minor deviation. Not sure if we really should change this until the magic numbers which are used are really clear (195, 193, 44, 36). Even though 44 and 36 I have good guesses ;)

File: XBMC Remote/KenBurns/JBKenBurnsView.m (hunk 4 above)
Comment (wutschel) [reply XBMC Remote/KenBurns/JBKenBurnsView.m]: Done with ff72a45.

File: XBMC Remote/iPad/StackScrollViewController.m (hunk 12 above)
Comment (wutschel) [reply XBMC Remote/iPad/StackScrollViewController.m]: For `NSTimeInterval` assignments done with 855e369.

File: XBMC Remote/DetailViewController.m [hunk 16]
```diff
@@ -1615,34 +1615,34 @@ - (void)indexViewValueChanged:(BDKCollectionIndexView *)sender {
...
-        [self alphaView:sectionNameOverlayView AnimDuration:0.3f Alpha:0];
+        [self alphaView:sectionNameOverlayView AnimDuration:0.3 Alpha:0];
     }
     _indexView.alpha = 1.0;
 }
 
 #pragma mark - Table Animation
 
--(void)alphaImage:(UIImageView *)image AnimDuration:(float)seconds Alpha:(float)alphavalue{
+-(void)alphaImage:(UIImageView *)image AnimDuration:(NSTimeInterval)seconds Alpha:(CGFloat)alphavalue{
```
[the remark this answers is not in the document]
Comment (wutschel) [reply XBMC Remote/DetailViewController.m:1629]: Agreed, but let us keep this out of this PR.

File: XBMC Remote/HostViewController.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/HostViewController.m:49]: Yes, same opinion here: separate PR.

File: XBMC Remote/NowPlaying.m (hunk 5 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m:273]: And again: separate PR.

File: XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m (hunk 9 above)
Comment (wutschel) [reply XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m:556]: I have no idea what to do with this.

File: XBMC Remote/DetailViewController.m (hunk 13 above)
[the remark this answers is not in the document]
Comment (kambala-decapitator) [reply XBMC Remote/DetailViewController.m]: looks pretty straightforward: just multiply size by `UIScreen.mainScreen.scale` and remove all conditions

File: XBMC Remote/PosterCell.m (hunk 6 above)
Comment (kambala-decapitator) [reply XBMC Remote/PosterCell.m]: please use the rounding function instead of this poor man's solution

File: XBMC Remote/DetailViewController.m (hunk 13 above)
[the remark this answers is not in the document]
Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: Melts down to 

`return CGSizeMake(size.width * UIScreen.mainScreen.scale, size.height * UIScreen.mainScreen.scale);`

File: XBMC Remote/RightMenuViewController.m (hunk 7 above)
Comment (kambala-decapitator) [reply XBMC Remote/RightMenuViewController.m:96]: at least extracting it to a separate function would be better, even better if you can figure out the explanation for such computation :)

File: XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m (hunk 9 above)
Comment (kambala-decapitator) [reply XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m:556]: the condition means "if iOS version is >= 5", hope it's clear what needs to be done ;)

File: XBMC Remote/DetailViewController.m (hunk 13 above)
[the remark this answers is not in the document]
Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: Done with b1b299c.

File: XBMC Remote/PosterCell.m (hunk 6 above)
Comment (wutschel) [reply XBMC Remote/PosterCell.m]: Done with de3e370.

I will have lots of fun squashing, and with rebasing to updated master ...

File: XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m (hunk 9 above)
Comment (wutschel) [reply XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m:556]: Ah, now I got it. Thought you were commenting on `alphaGradientLocations`.

File: XBMC Remote/iPad/StackScrollViewController.m (hunk 12 above)
Comment (kambala-decapitator) [reply XBMC Remote/iPad/StackScrollViewController.m]: > Question: There are still several cases where `floatValue` is used. I understand the variable type is relevant for the output (= the variable type the value assigned to). Is this understanding correct?

Yes, you're right, but in fact it depends on the floating-point value stored. `float` has simply less precision than `double`.

For example, if you simply feed `float` to formatting string, then you need to check how many digits after decimal point you want (for anything less than 7 `float` is ok).

In case of `resumePointPercentage` `float` is also fine because you always treat it as `float`.

Basically you need to review every case and decide whether `float` is sufficient or switch to `double`.

File: XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m (hunk 9 above)
Comment (wutschel) [reply XBMC Remote/SVPullToRefresh/UIScrollView+SVPullToRefresh.m]: Done with c4053b2.

File: XBMC Remote/iPad/StackScrollViewController.m (hunk 12 above)
Comment (wutschel) [reply XBMC Remote/iPad/StackScrollViewController.m]: I was thinking of all the width/height definition which are still done in `AppDelegate.m`, and the read access to it.

Comment (kambala-decapitator) [reply XBMC Remote/iPad/StackScrollViewController.m]: are you talking about https://github.com/xbmc/Official-Kodi-Remote-iOS/blob/master/XBMC%20Remote/AppDelegate.m#L95-L99 ?

Comment (wutschel) [reply XBMC Remote/iPad/StackScrollViewController.m]: Yes, there is lots of code like this: 

```
float itemMovieWidthLargeIpad =158.0f;
...
[NSNumber numberWithFloat:itemMovieWidthLargeIpad], @"width",
```

Comment (kambala-decapitator) [reply XBMC Remote/iPad/StackScrollViewController.m]: as long as those values are only wrapped in `NSNumber`, no need to change

Comment (wutschel) [reply XBMC Remote/iPad/StackScrollViewController.m]: Also the `float` declarations shall be kept?

Comment (kambala-decapitator) [reply XBMC Remote/iPad/StackScrollViewController.m]: yep

Comment (wutschel) [reply XBMC Remote/iPad/StackScrollViewController.m]: Ok, resolved then.

Comment (kambala-decapitator) [reply XBMC Remote/iPad/StackScrollViewController.m]: but you could change all those declarations to `NSNumber` directly if you wish, e.g.
```objc
__auto_type itemMovieWidthLargeIpad = @(158);
```

**Comment by kambala-decapitator** [comment]
please solve the last point, resolve conflicts and LGTM

File: XBMC Remote/RightMenuViewController.m (hunk 7 above)
Comment (kambala-decapitator) [reply XBMC Remote/RightMenuViewController.m:96]: this seems pending

Comment (wutschel) [reply XBMC Remote/RightMenuViewController.m:96]: Yes, on it. Try to figure our what this really is about. Would great if you could check if there is anything missing for PR https://github.com/xbmc/Official-Kodi-Remote-iOS/pull/139. From an end user point of view that´s having more impact.

Comment (wutschel) [reply XBMC Remote/RightMenuViewController.m]: So, 18c109d does the trick. The logo is nearly invisible, but the calculation is now understood. If you´re ok with this, I will look into squashing some of the commits and rebasing.

**Comment by wutschel** [comment]
Squashed and rebased to master.
